src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -50,6 +50,26 @@ namespace
         ::DeregisterEventSource(eventSource);
     }
 
+    bool try_get_url_from_line(const pal::string_t& line, pal::string_t& url)
+    {
+        const pal::string_t url_prefix = DOTNET_CORE_APPLAUNCH_URL _X("?");
+        if (starts_with(line, url_prefix, true))
+        {
+            url.assign(line);
+            return true;
+        }
+
+        const pal::string_t url_prefix_before_7_0 = _X("  - ") DOTNET_CORE_APPLAUNCH_URL _X("?");
```

Comment:
This is technically an allocation. Since it is also called in a loop, we are allocating on each call. This type and all uses can be updated to use compile-time constants. For example, the length and all uses could be converted to `const pal::char_t*`.

Reply:
Ugh, yeah, I left it as a string since that's what `starts_with` and `ends_with` take and this is only for specific errors on GUI apps. I didn't feel like updating those utils for this change, but I can go try updating them.